### ISSUE TITLE
[bug] npm version patch errors with "npm ERR! stderr maxBuffer exceeded."

### ISSUE THREAD

**Issue by koulmomo** [npm/npm#11670]
like the title says, when running `npm version --verbose patch`, the following (redacted) is printed.

```sh
foo git:(master) npm version --verbose patch
npm info it worked if it ends with ok
npm verb cli [ 'node',
npm verb cli   '/Users/koulmomo/.nvm/v0.10.26/bin/npm',
npm verb cli   'version',
npm verb cli   '--verbose',
npm verb cli   'patch' ]
npm info using npm@3.7.3
npm info using node@v0.10.26
npm info git [ 'status', '--porcelain' ]
npm info lifecycle foo@3.2.1~preversion: foo@3.2.1
npm verb version.write data { name: 'foo',
npm verb version.write   version: '3.2.1',
npm verb version.write   author: 'Mo Kouli <mkouli@fake.com>'...}
v3.2.1
npm info lifecycle foo@3.2.1~version: foo@3.2.1
npm info git [ 'add',
npm info git   '/Users/koulmomo/foo/package.json' ]
npm info git [ 'commit', '-m', '3.2.1' ]
npm verb stack Error: stderr maxBuffer exceeded.
npm verb stack     at Socket.<anonymous> (child_process.js:686:13)
npm verb stack     at Socket.EventEmitter.emit (events.js:95:17)
npm verb stack     at Socket.<anonymous> (_stream_readable.js:746:14)
npm verb stack     at Socket.EventEmitter.emit (events.js:92:17)
npm verb stack     at emitReadable_ (_stream_readable.js:408:10)
npm verb stack     at emitReadable (_stream_readable.js:404:5)
npm verb stack     at readableAddChunk (_stream_readable.js:165:9)
npm verb stack     at Socket.Readable.push (_stream_readable.js:127:10)
npm verb stack     at Pipe.onread (net.js:528:21)
npm verb cwd /Users/koulmomo/foo
npm ERR! Darwin 15.3.0
npm ERR! argv "node" "/Users/koulmomo/.nvm/v0.10.26/bin/npm" "version" "--verbose" "patch"
npm ERR! node v0.10.26
npm ERR! npm  v3.7.3

npm ERR! stderr maxBuffer exceeded.
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/koulmomo/foo/npm-debug.log
```

**Comment by ganemone** [comment]
npm should probably should use spawn instead of exec so you can pipe stdout/stderr and not buffer it all in memory.

**Comment by koulmomo** [comment]
this is no longer happening, not sure where/when the fix was but working for me with `npm@3.9.0`.

**Comment by jindrahm** [comment]
I had the same problem.. solved by disabling logging of some gulp tasks.
"npm version patch" does commit and I have some pre-commit hooks to build and test my app.. but the build log was too big for the buffer.
